main.go

```diff
@@ -1,22 +1,47 @@
 package main
 
 import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	log "github.com/sirupsen/logrus"
 	"net/http"
 	"os"
 )
 
 func indexHandler(w http.ResponseWriter, r *http.Request) {
-	w.Write([]byte("<h1>Hello Liatrio!</h1>"))
+	_, err := w.Write([]byte("<h1>Hello Liatrio!</h1>"))
+
+	log.Debugf("Request received on %s", r.RequestURI)
+
+	if err != nil {
+		log.Errorf("error sending response: %s", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	hellosGiven.Inc()
 }
 
+var (
+	hellosGiven = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "cloud_native_go_helloworld_service_hellos_given_total",
+		Help: "The total number of hellos returned by the service",
+	})
+)
+
 func main() {
 	port := os.Getenv("PORT")
 	if port == "" {
 		port = "3000"
 	}
 
-	mux := http.NewServeMux()
+	log.SetLevel(log.DebugLevel)
```

Comment:
I dropped this for local dev to see request logging, might need to be made more dynamic in the future